File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    bucket = []
    cnt = 0
    moves = list(map(lambda x: x - 1, moves))
    
    # 연산하기 쉽게 보드판 새로 짜기
    new_board = []
    for i in range(len(board)):
        tmp = []
        for j in range(len(board)):
            tmp.append(board[j][i]) 
        new_board.append(tmp)    
    
    # 인형뽑기 실행
    for move in moves:
        for i in range(len(new_board[move])):
            if new_board[move][i] != 0:
                if len(bucket) > 0 and bucket[-1] == new_board[move][i]:
                    bucket.pop()
                    cnt += 2
                else:
                    bucket.append(new_board[move][i])
                new_board[move][i] = 0
                break

    return cnt

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>(); 
        int n = board.length;
        int[] reMoves = Arrays.stream(moves).map(m -> m - 1).toArray();
        
        for (int m : reMoves) {
            for (int i = 0; i < n; i++) {
                // 0이 아니면, 해당 숫자를 얻는다.
                int doll = board[i][m];
                
                if (doll != 0) {
                // 스택 통이 비어있지 않고, 안에 peek값고 해당 숫자가 같다면, answer += 2를 하고,
                // 스택 통을 pop한다.
                    if (!stack.isEmpty() && stack.peek() == doll) {
                        answer += 2;
                        stack.pop();
                    } else {
                        // 그렇지 않다면 해당 숫자를 push한다.
                        stack.push(doll);
                    }
                board[i][m] = 0;
                break;
                }
            }
        }
        
        return answer;
    }
}